test/integration/good/compiler-optimizations/cpp.expected

```diff
@@ -365,298 +311,768 @@ class optimizations_model : public model_base_crtp<optimizations_model> {
-          current_statement__ = 62;
-          lp_accum__.add(y);
+          current_statement__ = 44;
+          lp_accum__.add(24);
         }
         {
-          local_scalar_t__ y;
-          
-          current_statement__ = 64;
-          y = 22;
-          current_statement__ = 65;
-          y = 245;
-          current_statement__ = 66;
-          lp_accum__.add(y);
-        }
-        current_statement__ = 69;
-        if (0) {
-          current_statement__ = 68;
-          lp_accum__.add(235);
-        } 
-        current_statement__ = 71;
-        if (1) {
-          current_statement__ = 70;
-          lp_accum__.add(2);
-        } 
-        current_statement__ = 74;
-        if ((24 * 2)) {
-          
-        } else {
+          double y;
           
+          current_statement__ = 47;
+          lp_accum__.add(245);
         }
-        current_statement__ = 77;
-        if ((24 * 2)) {
-          
-        } else {
-          
+        {
+          {
+            current_statement__ = 49;
+            lp_accum__.add(2);
+          }
         }
-        current_statement__ = 79;
-        if ((24 * 2)) {
-          
-        } 
-        current_statement__ = 81;
-        if ((20 * 2)) {
-          
-        } 
-        current_statement__ = 84;
-        if (rfun_lp(lp__, lp_accum__, pstream__)) {
-          
-        } else {
-          
+        int sym22__;
+        for (size_t sym21__ = 1; sym21__ <= 1; ++sym21__) {
+          {
+            current_statement__ = 51;
+            lp_accum__.add(2);
+            break;
+          }}
+        current_statement__ = 54;
+        while (576) {
+          {
+            break;
+          }
         }
-        current_statement__ = 87;
-        while (0) {
-          current_statement__ = 85;
-          lp_accum__.add(325);
-        }
-        current_statement__ = 89;
-        while ((24 * 24)) {
-          break;
-        }
-        current_statement__ = 91;
-        while (rfun_lp(lp__, lp_accum__, pstream__)) {
-          break;
-        }
-        current_statement__ = 93;
-        for (size_t i = 31; i <= 225; ++i) { continue;}
-        current_statement__ = 95;
-        for (size_t i = 31; i <= 225; ++i) { break;}
-        current_statement__ = 97;
-        for (size_t i = 31; i <= 225; ++i) { }
-        current_statement__ = 99;
-        for (size_t i = rfun_lp(lp__, lp_accum__, pstream__); i <= 225; ++i) {
-          continue;}
-        current_statement__ = 101;
-        for (size_t i = rfun_lp(lp__, lp_accum__, pstream__); i <= 225; ++i) {
-          break;}
-        current_statement__ = 103;
-        for (size_t i = rfun_lp(lp__, lp_accum__, pstream__); i <= 225; ++i) {
+        int sym25__;
+        for (size_t sym24__ = 1; sym24__ <= 1; ++sym24__) {
+          {
+            current_statement__ = 51;
+            lp_accum__.add(2);
+            sym25__ = 24;
+            break;
+          }}
+        while (sym25__) {
+          {
+            {
+              break;
```

Comment:
what is happening here??

Reply:
Not sure. I'll see if putting dead code elimination at the end would fix this.